### ISSUE TITLE
[multi-runner] Can't launch runners with a 4th (custom) label

### ISSUE THREAD

**Issue by alpozcan** [philips-labs/terraform-aws-github-runner#3420]
Hi,
I have the below code making use of the `multi-runner` module but I could never get the non-default runner configurations  launched. Below is in the `locals`:
```
  runner_configs = {
    "arm-default" = { arch = "arm64", instance_class = "m6g", instance_size = "xlarge" }
    "x86-default" = { arch = "x64", instance_class = "m6a", instance_size = "xlarge" }

    # couldn't ever get the following specialised ones working
    "x86-small"  = { arch = "x64", instance_class = "m6a", instance_size = "small" }
    "x86-medium" = { arch = "x64", instance_class = "m6a", instance_size = "medium" }
```

which are then fed into
```
  multi_runner_config = { for name, config in local.runner_configs :
    name => {
      matcherConfig = {
        labelMatchers = [
          endswith(name, "-default") ? ["self-hosted", "linux", config.arch] : ["self-hosted", "linux", config.arch, config.instance_size]
        ]
        exactMatch = true
      }
      runner_config = merge(
        local.runner_config_common,
        {
          instance_types      = ["${config.instance_class}.${config.instance_size}"]
          runner_architecture = config.arch
          runner_extra_labels = join(",", [config.arch, config.instance_size, name])
      })
    }
  }
```

I have already tried different ordering of the configs, different labels, using extra labels, etc. No matter what, it always launches the `-default` config and if I add a 4th label such as `small`, the jobs never get picked up by a worker - no worker gets launched at all.

In other words, if in my workflow YAML I have requested the runners like `runs-on: [self-hosted, linux, x64]`, it works and runs in an `xlarge`. But if I try with e.g. `runs-on: [self-hosted, linux, x64, small]`, no runner gets launched.

**Comment by alpozcan** [comment]
turns out it was due to unavailability of particular instance type in the region.
